site/docs/build-ref.html

```diff
@@ -167,82 +168,78 @@ <h3 id="labels">Labels</h3>
-  the colon is omitted, the target name is assumed to be the same as
-  the last component of the package name, so these two labels are
-  equivalent:
+  The second part of the label is the un-qualified package name
+  <code>my/app/main</code>, the path to the package
+  relative to the repository root.  When the label refers to the same
+  package it is used in, the package name (and optionally, the colon)
+  may be omitted.  So, inside <code>@myrepo//my/app/main</code>,
+  this label may be written either of the following ways:
 </p>
 
 <pre>
-//my/app
-//my/app:app
+app_binary
+:app_binary
 </pre>
 
 <p>
-  Short-form labels such as <code>//my/app</code> are not to
-  be confused with package names. Labels start with <code>//</code>,
-  but package names never do, thus <code>my/app</code> is the
-  package containing <code>//my/app</code>.
-
-  (A common misconception is that <code>//my/app</code> refers
-  to a package, or to <em>all</em> the targets in a package; neither
-  is true.)
+  (It is a matter of convention that the colon is omitted for files,
+  but retained for rules, but it is not otherwise significant.)
 </p>
 
 <p>
-  Within a BUILD file, the package-name part of label may be omitted,
-  and optionally the colon too. So within the BUILD file for package
-  <code>my/app</code> (i.e. <code>//my/app:BUILD</code>),
-  the following "relative" labels are all equivalent:
+  The part of the label after
+  the colon, <code>app_binary</code> is the un-qualified target name.
+  When it matches the last component of the package path, it, and
+  the colon, may be omitted.  So, these two labels are 
+  equivalent:
 </p>
 
 <pre>
-//my/app:app
-//my/app
-:app
-app
+//my/app/lib
+//my/app/lib:lib
 </pre>
 
 <p>
-  (It is a matter of convention that the colon is omitted for files,
-  but retained for rules, but it is not otherwise significant.)
+  The name of a file target in a subdirectory of the
+  package is the file's path relative to the package
+  root (the directory containing the <code>BUILD</code> file),
+  so this file is in the <code>my/app/main/testdata</code>
+  subdirectory of the repository:
 </p>
 
-<p>
-  Similarly, within a BUILD file, files belonging to the package may
-  be referenced by their unadorned name relative to the package
-  directory:
-</p>
-
-
 <pre>
-generate.cc
-testdata/input.txt
+//my/app/main:testdata/input.txt
 </pre>
 
 <p>
-  But from other packages, or from the command-line, these file
-  targets must always be referred to by their complete label, e.g.
-  <code>//my/app:generate.cc</code>.
+  Labels such as <code>//my/app</code> are not to
+  be confused with package names. Labels always start with a
+  repository identifier (often abbreviated <code>//</code>)
+  but package names never do. Thus, <code>my/app</code> is the
+  package containing <code>//my/app/lib</code> (a.k.a. 
+  <code>//my/app/lib:lib</code>).
+
+  (A common misconception is that <code>//my/app</code> refers
```

Comment:
Is it worth explaining what it *does* mean? Or position it in the text where that implication is clear?